include/nlohmann/json.hpp

```diff
@@ -3607,16 +3607,10 @@ class basic_json
                 // remember array size before resizing
                 const auto previous_size = m_value.array->size();
 #endif
-
                 m_value.array->resize(idx + 1);
 
-#if JSON_DIAGNOSTICS
                 // set parent for values added above
-                for (auto i = previous_size; i <= idx; ++i)
-                {
-                    set_parent(m_value.array->operator[](i));
-                }
-#endif
+                set_parents(begin() + previous_size, idx + 1 - previous_size);
```

Comment:
Since you are now always using `previous_size` you need to remove the `#if/#endif` above.

Reply:
...or you can add them back in as you did in the next commit.  :)